src/main/java/org/jenkinsci/test/acceptance/po/Login.java

```diff
@@ -40,7 +41,7 @@ public Login doLogin(String user, String password){
     public Login doLoginDespiteNoPaths(String user, String password){
         driver.findElement(by.name("j_username")).sendKeys(user);
         driver.findElement(by.name("j_password")).sendKeys(password);
-        clickButton("log in");
+        driver.findElement(by.name("Submit")).click();
```

Comment:
I tend to use `waitFor` to shield against performance issues instead of `findElement`

Reply:
Well the goal was just to avoid finding element by text but by name and do what clickButton is doing (would have done clickButton(by.name("...") if possible): https://github.com/jenkinsci/acceptance-test-harness/blob/5e551aa8b82398accdd45366add77bbaa08a98b5/src/main/java/org/jenkinsci/test/acceptance/po/CapybaraPortingLayerImpl.java#L113

I tend to think that the wait for should be done at the page level and not for each element, no real strong opinion on that, open to other feedback, I can add it if needed.

Reply:
I suggest not to pollute the code with unnecessary waits. Let's wait until it turns necessary.